lib/src/test/java/dev/openfeature/javasdk/DoSomethingProvider.java

```diff
@@ -4,12 +4,7 @@ public class DoSomethingProvider implements FeatureProvider {
 
     @Override
     public Metadata getMetadata() {
-        return new Metadata() {
-            @Override
-            public String getName() {
-                return "test";
-            }
-        };
+        return () -> "test";
```

Comment:
I'm not familiar with this language feature. Do you know the name of it?

Reply:
When a class contains a single method, it's called a functional interface. Because it contains only a single method, it can be expressed as a single lambda. Basically this class can be implemented by returning a single higher order function.

I think they were introduced in Java 8